### PR TITLE
Automated tier1 and tier2 tests for compute resource rhev

### PR DESCRIPTION
**Pull request by san7ket**
nosetests tests.foreman.ui.test_computeresource_rhev
```
..S.....SSSS...
----------------------------------------------------------------------
Ran 15 tests in 615.541s

OK (SKIP=5)
```

### CODE REVIEW

File: tests/foreman/ui/test_computeresource_rhev.py [hunk 1]
```diff
@@ -200,12 +344,38 @@ def test_negative_add_image_rhev_with_invalid_name(self):
...
+                                  parameter_list=parameter_list
+                                  )
+                parameter_list1 = [
+                           ['Name', img_name],
+                           ['Operatingsystem', self.rhev_img_os],
+                           ['Architecture', self.rhev_img_arch],
+                           ['Username', self.rhev_img_user],
+                           ['Password', self.rhev_img_pass],
+                           ['uuid', self.rhev_img_name],
+                           ]
```
Comment (kbidarkar) [inline tests/foreman/ui/test_computeresource_rhev.py]: Please move this list assignment above,

**Comment by kbidarkar** [comment]
Also let's define the lists as below,

```
parameter_list = [
    ['URL', self.rhev_url, 'field'],
    ['Username', self.rhev_username, 'field'],
    ['Password', self.rhev_password, 'field'],
    ['Datacenter', self.rhev_datacenter, 'special select'],
]

```

File: tests/foreman/ui/test_computeresource_rhev.py (hunk 1 above)
Comment (san7ket) [inline tests/foreman/ui/test_computeresource_rhev.py]: Need to take img_name from the invalid_names_list(),If it goes above loop, it is referenced before assignment
Anyway to solve this?

File: robottelo/ui/locators/base.py [hunk 2]
```diff
@@ -245,7 +245,9 @@
         "//a[contains(@data-url, '/compute_resources/test_connection')]"),
     "resource.username": (By.ID, "compute_resource_user"),
     "resource.password": (By.ID, "compute_resource_password"),
-    "resource.datacenter": (By.XPATH, "//select[@id='compute_resource_uuid']"),
+    "resource.datacenter": (By.XPATH, "//div[contains"
+                            "(@id, 's2id_compute_resource_uuid')]"
+                            "/a/span[contains(@class, 'arrow')]"),
```
Comment (kbidarkar) [inline robottelo/ui/locators/base.py]: Also, let's update the locator as below for consistency sake and for better readability,

```
"resource.datacenter": (
    By.XPATH, 
    ("//div[contains(@id, 's2id_compute_resource_uuid')]"
     "/a/span[contains(@class, 'arrow')]")),
```

File: tests/foreman/ui/test_computeresource_rhev.py [hunk 3]
```diff
@@ -61,12 +98,27 @@ def test_positive_create_rhev_with_description(self):
...
+        parameter_list = [
+                        ['URL', self.rhev_url, 'field'],
+                        ['Username', self.rhev_username, 'field'],
+                        ['Password', self.rhev_password, 'field'],
+                        ['Datacenter', self.rhev_datacenter, 'special select']
+                        ]
+        with Session(self.browser) as session:
+            for description in valid_data_list():
+                with self.subTest(description):
+                    name = gen_string('alpha')
```
Comment (kbidarkar) [inline tests/foreman/ui/test_computeresource_rhev.py]: Let's define this before starting a browser session.

File: tests/foreman/ui/test_computeresource_rhev.py [hunk 4]
```diff
@@ -10,18 +10,41 @@
...
+from robottelo.config import settings
+from robottelo.constants import FOREMAN_PROVIDERS
+from robottelo.datafactory import invalid_names_list, valid_data_list
 from robottelo.decorators import run_only_on, tier1, tier2, tier3, stubbed
+from fauxfactory import gen_string
+from nailgun import entities
 from robottelo.test import UITestCase
+from robottelo.ui.factory import make_resource
+from robottelo.ui.session import Session
+from robottelo.ui.locators import common_locators
```
Comment (kbidarkar) [inline tests/foreman/ui/test_computeresource_rhev.py]: Let's change the import as per alphabetic order.

File: tests/foreman/ui/test_computeresource_rhev.py (hunk 1 above)
Comment (kbidarkar) [inline tests/foreman/ui/test_computeresource_rhev.py]: oh! , then it is ok.

File: robottelo/ui/locators/base.py [hunk 5]
```diff
@@ -320,14 +322,17 @@
...
+        "/a/span[contains(@class, 'arrow')]"),
     "resource.image_architecture": (
-        By.XPATH, "//select[@id='image_architecture_id']"),
+        By.XPATH, "//div[@id='s2id_image_architecture_id']"
+        "/a/span[contains(@class, 'arrow')]"),
     "resource.image_username": (By.ID, "image_username"),
     "resource.image_password": (By.ID, "image_password"),
-    "resource.image_uuid": (By.XPATH, "//input[@id='image_uuid']"),
+    "resource.image_uuid": (By.XPATH, "//div[@id='s2id_image_uuid']"
+                            "/a/span[contains(@class, 'arrow')]"),
```
Comment (sghai) [inline robottelo/ui/locators/base.py]:
```python
"resource.image_uuid": (
        By.XPATH, 
        "//div[@id='s2id_image_uuid']/a/span[contains(@class, 'arrow')]"),
```

File: tests/foreman/ui/test_computeresource_rhev.py [hunk 6]
```diff
@@ -177,12 +298,37 @@ def test_positive_add_image_rhev_with_name(self):
...
+        name = gen_string('alpha')
+        with Session(self.browser) as session:
+            for img_name in valid_data_list():
+                with self.subTest(name):
+                    make_resource(
+                                  session,
+                                  name=name,
+                                  provider_type=FOREMAN_PROVIDERS['rhev'],
+                                  parameter_list=parameter_list
+                                  )
```
Comment (sghai) [inline tests/foreman/ui/test_computeresource_rhev.py]: `make resource` not properly indented.

File: tests/foreman/ui/test_computeresource_rhev.py [hunk 7]
```diff
@@ -200,12 +346,38 @@ def test_negative_add_image_rhev_with_invalid_name(self):
...
+        name = gen_string('alpha')
+        with Session(self.browser) as session:
+            for img_name in invalid_names_list():
+                with self.subTest(name):
+                    make_resource(
+                                  session,
+                                  name=name,
+                                  provider_type=FOREMAN_PROVIDERS['rhev'],
+                                  parameter_list=parameter_list
+                                  )
```
Comment (sghai) [inline tests/foreman/ui/test_computeresource_rhev.py]: make_resource not properly indented.

File: tests/foreman/ui/test_computeresource_rhev.py [hunk 8]
```diff
@@ -107,12 +176,30 @@ def test_positive_update_rhev_name(self):
...
+            newname = gen_string('alpha')
+            with self.subTest(newname):
+                name = gen_string('alpha')
+                make_resource(
+                    session,
+                    name=name,
+                    provider_type=FOREMAN_PROVIDERS['rhev'],
+                    parameter_list=parameter_list
+                )
+                search = self.compute_resource.search(name)
```
Comment (oshtaier) [inline tests/foreman/ui/test_computeresource_rhev.py]: Don't need to use a variable here, make an assertion right away
self.assertIsNotNone(self.compute_resource.search(name))

File: tests/foreman/ui/test_computeresource_rhev.py [hunk 9]
```diff
@@ -107,12 +176,30 @@ def test_positive_update_rhev_name(self):
...
+                self.assertIsNotNone(search)
+                self.compute_resource.update(name=name, newname=newname)
+                search = self.compute_resource.search(newname)
```
Comment (oshtaier) [inline tests/foreman/ui/test_computeresource_rhev.py]: Don't need to use a variable here, make an assertion right away
`self.assertIsNotNone(self.compute_resource.search(name))`

File: tests/foreman/ui/test_computeresource_rhev.py [hunk 10]
```diff
@@ -107,12 +176,30 @@ def test_positive_update_rhev_name(self):
         5. Update the name of the created CR with valid string.
 
         @Assert: The rhev Compute Resource is updated
-
-        @Caseautomation: notautomated
         """
+        parameter_list = [
```
Comment (oshtaier) [inline tests/foreman/ui/test_computeresource_rhev.py]: it doesn't seem as properly intended list:
```
parameter_list = [
                               ['URL', self.rhev_url, 'field'],
                               ['Username', self.rhev_username, 'field'],
                               ['Password', self.rhev_password, 'field'],
                               ['Datacenter', self.rhev_datacenter, 'special select']
]
```

File: tests/foreman/ui/test_computeresource_rhev.py [hunk 11]
```diff
@@ -131,12 +218,32 @@ def test_positive_update_rhev_organization(self):
         6. Add the new CR to organization that is created.
 
         @Assert: The rhev Compute Resource is updated
-
-        @Caseautomation: notautomated
         """
+        parameter_list = [
```
Comment (oshtaier) [inline tests/foreman/ui/test_computeresource_rhev.py]: same here

File: tests/foreman/ui/test_computeresource_rhev.py [hunk 12]
```diff
@@ -131,12 +218,32 @@ def test_positive_update_rhev_organization(self):
...
+        with Session(self.browser) as session:
+            make_resource(
+                session,
+                name=name,
+                provider_type=FOREMAN_PROVIDERS['rhev'],
+                parameter_list=parameter_list,
+                orgs=[entities.Organization().create().name],
+                org_select=True
+            )
+            search = self.compute_resource.search(name)
```
Comment (oshtaier) [inline tests/foreman/ui/test_computeresource_rhev.py]: `self.assertIsNotNone(self.compute_resource.search(name))`

File: tests/foreman/ui/test_computeresource_rhev.py [hunk 13]
```diff
@@ -154,12 +261,26 @@ def test_positive_delete_rhev(self):
         5. Delete the created compute resource.
 
         @Assert: The Compute Resource is deleted
-
-        @Caseautomation: notautomated
         """
+        parameter_list = [
```
Comment (oshtaier) [inline tests/foreman/ui/test_computeresource_rhev.py]: same here

File: tests/foreman/ui/test_computeresource_rhev.py [hunk 14]
```diff
@@ -177,12 +298,37 @@ def test_positive_add_image_rhev_with_name(self):
         5. Select the desired template to create image and submit.
 
         @Assert: The image is added to the CR successfully
-
-        @Caseautomation: notautomated
          """
+        parameter_list = [
```
Comment (oshtaier) [inline tests/foreman/ui/test_computeresource_rhev.py]: same here

File: tests/foreman/ui/test_computeresource_rhev.py [hunk 15]
```diff
@@ -200,12 +346,38 @@ def test_negative_add_image_rhev_with_invalid_name(self):
         5. Select the desired template to create the image from and submit.
 
         @Assert: The image should not be added to the CR
-
-        @Caseautomation: notautomated
         """
+        parameter_list = [
```
Comment (oshtaier) [inline tests/foreman/ui/test_computeresource_rhev.py]: same here

File: tests/foreman/ui/test_computeresource_rhev.py [hunk 16]
```diff
@@ -223,9 +395,23 @@ def test_positive_access_rhev_with_default_profile(self):
         5. Select (3-Large) and submit.
 
         @Assert: The Compute Resource created and opened successfully
-
-        @Caseautomation: notautomated
         """
+        parameter_list = [
```
Comment (oshtaier) [inline tests/foreman/ui/test_computeresource_rhev.py]: same here

File: tests/foreman/ui/test_computeresource_rhev.py [hunk 17]
```diff
@@ -266,9 +451,25 @@ def test_positive_retrieve_rhev_vm_list(self):
         2. Go to "Virtual Machines" tab.
 
         @Assert: The Virtual machines should be displayed
-
-        @Caseautomation: notautomated
         """
+        parameter_list = [
```
Comment (oshtaier) [inline tests/foreman/ui/test_computeresource_rhev.py]: and here

File: tests/foreman/ui/test_computeresource_rhev.py [hunk 18]
```diff
@@ -266,9 +451,25 @@ def test_positive_retrieve_rhev_vm_list(self):
...
+                          ]
+        name = gen_string('alpha')
+        with Session(self.browser) as session:
+            make_resource(
+                session,
+                name=name,
+                provider_type=FOREMAN_PROVIDERS['rhev'],
+                parameter_list=parameter_list
+            )
+            search = self.compute_resource.search(name)
```
Comment (oshtaier) [inline tests/foreman/ui/test_computeresource_rhev.py]: same about variable

File: tests/foreman/ui/test_computeresource_rhev.py [hunk 19]
```diff
@@ -38,13 +59,29 @@ def test_positive_create_rhev_with_name(self):
         3. Provide a valid name to rhev Compute Resource.
         4. Test the connection using Load Datacenter and submit.
 
-        @Assert: A rhev CR is created successfully with proper connection.
+        @CaseAutomation: Automated
 
-        @Caseautomation: notautomated
+        @Assert: A rhev CR is created successfully with proper connection.
         """
+        parameter_list = [
```
Comment (oshtaier) [inline tests/foreman/ui/test_computeresource_rhev.py]: Once more:
```
```
parameter_list = [
    ['URL', self.rhev_url, 'field'],
    ['Username', self.rhev_username, 'field'],
    ['Password', self.rhev_password, 'field'],
    ['Datacenter', self.rhev_datacenter, 'special select'],
]

File: tests/foreman/ui/test_computeresource_rhev.py [hunk 20]
```diff
@@ -38,13 +59,29 @@ def test_positive_create_rhev_with_name(self):
...
+                          ['URL', self.rhev_url, 'field'],
+                          ['Username', self.rhev_username, 'field'],
+                          ['Password', self.rhev_password, 'field'],
+                          ['Datacenter', self.rhev_datacenter,
+                           'special select'],
+                          ]
+        with Session(self.browser) as session:
+            for name in valid_data_list():
+                with self.subTest(name):
+                    make_resource(
```
Comment (oshtaier) [inline tests/foreman/ui/test_computeresource_rhev.py]:
```
make_resource(
    session,
    name=name,
    provider_type=FOREMAN_PROVIDERS['rhev'],
    parameter_list=parameter_list
)
```

File: tests/foreman/ui/test_computeresource_rhev.py [hunk 21]
```diff
@@ -60,13 +97,30 @@ def test_positive_create_rhev_with_description(self):
         3. Provide it with some valid description to rhev Compute Resource.
         4. Test the connection using Load Datacenter and submit.
 
-        @Assert: A rhev Compute Resource is created successfully
+        @CaseAutomation: Automated
 
-        @Caseautomation: notautomated
+        @Assert: A rhev Compute Resource is created successfully
         """
+        parameter_list = [
```
Comment (oshtaier) [inline tests/foreman/ui/test_computeresource_rhev.py]:
```
parameter_list = [
    ['URL', self.rhev_url, 'field'],
    ['Username', self.rhev_username, 'field'],
    ['Password', self.rhev_password, 'field'],
    ['Datacenter', self.rhev_datacenter, 'special select'],
]
```

File: tests/foreman/ui/test_computeresource_rhev.py [hunk 22]
```diff
@@ -60,13 +97,30 @@ def test_positive_create_rhev_with_description(self):
...
+                          ['Username', self.rhev_username, 'field'],
+                          ['Password', self.rhev_password, 'field'],
+                          ['Datacenter', self.rhev_datacenter,
+                           'special select'],
+                          ]
+        name = gen_string('alpha')
+        with Session(self.browser) as session:
+            for description in valid_data_list():
+                with self.subTest(description):
+                    make_resource(
```
Comment (oshtaier) [inline tests/foreman/ui/test_computeresource_rhev.py]:
```
make_resource(
    session,
    name=name,
    provider_type=FOREMAN_PROVIDERS['rhev'],
    parameter_list=parameter_list
)
```

File: tests/foreman/ui/test_computeresource_rhev.py [hunk 23]
```diff
@@ -83,13 +137,33 @@ def test_negative_create_rhev_with_invalid_name(self):
         3. Provide a invalid name to rhev Compute Resource.
         4. Test the connection using Load Datacenter and submit.
 
-        @Assert: A rhev Compute Resource is not created
+        @CaseAutomation: Automated
 
-        @Caseautomation: notautomated
+        @Assert: A rhev Compute Resource is not created
         """
+        parameter_list = [
```
Comment (oshtaier) [inline tests/foreman/ui/test_computeresource_rhev.py]: and here

File: tests/foreman/ui/test_computeresource_rhev.py [hunk 24]
```diff
@@ -83,13 +137,33 @@ def test_negative_create_rhev_with_invalid_name(self):
...
+                          ['URL', self.rhev_url, 'field'],
+                          ['Username', self.rhev_username, 'field'],
+                          ['Password', self.rhev_password, 'field'],
+                          ['Datacenter', self.rhev_datacenter,
+                           'special select'],
+                          ]
+        with Session(self.browser) as session:
+            for name in invalid_names_list():
+                with self.subTest(name):
+                    make_resource(
```
Comment (oshtaier) [inline tests/foreman/ui/test_computeresource_rhev.py]: and here

File: tests/foreman/ui/test_computeresource_rhev.py [hunk 25]
```diff
@@ -106,13 +180,32 @@ def test_positive_update_rhev_name(self):
         4. Test the connection using Load Datacenter and submit.
         5. Update the name of the created CR with valid string.
 
-        @Assert: The rhev Compute Resource is updated
+        @CaseAutomation: Automated
 
-        @Caseautomation: notautomated
+        @Assert: The rhev Compute Resource is updated
         """
+        parameter_list = [
```
Comment (oshtaier) [inline tests/foreman/ui/test_computeresource_rhev.py]: and here

File: tests/foreman/ui/test_computeresource_rhev.py [hunk 26]
```diff
@@ -106,13 +180,32 @@ def test_positive_update_rhev_name(self):
...
+                          ['Username', self.rhev_username, 'field'],
+                          ['Password', self.rhev_password, 'field'],
+                          ['Datacenter', self.rhev_datacenter,
+                           'special select'],
+                          ]
+        newname = gen_string('alpha')
+        name = gen_string('alpha')
+        with Session(self.browser) as session:
+            with self.subTest(newname):
+                make_resource(
```
Comment (oshtaier) [inline tests/foreman/ui/test_computeresource_rhev.py]: and here

File: tests/foreman/ui/test_computeresource_rhev.py [hunk 27]
```diff
@@ -130,13 +223,35 @@ def test_positive_update_rhev_organization(self):
...
+        parameter_list = [
+                          ['URL', self.rhev_url, 'field'],
+                          ['Username', self.rhev_username, 'field'],
+                          ['Password', self.rhev_password, 'field'],
+                          ['Datacenter', self.rhev_datacenter,
+                           'special select'],
+                          ]
+        name = gen_string('alpha')
+        with Session(self.browser) as session:
+            make_resource(
```
Comment (oshtaier) [inline tests/foreman/ui/test_computeresource_rhev.py]: and here

File: tests/foreman/ui/test_computeresource_rhev.py [hunk 28]
```diff
@@ -153,13 +268,30 @@ def test_positive_delete_rhev(self):
         4. Test the connection using Load Datacenter and submit.
         5. Delete the created compute resource.
 
-        @Assert: The Compute Resource is deleted
+        @CaseAutomation: Automated
 
-        @Caseautomation: notautomated
+        @Assert: The Compute Resource is deleted
         """
+        parameter_list = [
```
Comment (oshtaier) [inline tests/foreman/ui/test_computeresource_rhev.py]: and here

File: tests/foreman/ui/test_computeresource_rhev.py [hunk 29]
```diff
@@ -153,13 +268,30 @@ def test_positive_delete_rhev(self):
...
+                          ['Username', self.rhev_username, 'field'],
+                          ['Password', self.rhev_password, 'field'],
+                          ['Datacenter', self.rhev_datacenter,
+                           'special select'],
+                          ]
+        name = gen_string('alpha')
+        with Session(self.browser) as session:
+            with self.subTest(name):
+                make_resource(
+                              session,
```
Comment (oshtaier) [inline tests/foreman/ui/test_computeresource_rhev.py]: and here

File: tests/foreman/ui/test_computeresource_rhev.py [hunk 30]
```diff
@@ -176,13 +308,41 @@ def test_positive_add_image_rhev_with_name(self):
         4. Select "New image" , provide it valid name and information.
         5. Select the desired template to create image and submit.
 
-        @Assert: The image is added to the CR successfully
+        @CaseAutomation: Automated
 
-        @Caseautomation: notautomated
+        @Assert: The image is added to the CR successfully
          """
+        parameter_list = [
```
Comment (oshtaier) [inline tests/foreman/ui/test_computeresource_rhev.py]: and here

File: tests/foreman/ui/test_computeresource_rhev.py [hunk 31]
```diff
@@ -176,13 +308,41 @@ def test_positive_add_image_rhev_with_name(self):
...
+                          ['Username', self.rhev_username, 'field'],
+                          ['Password', self.rhev_password, 'field'],
+                          ['Datacenter', self.rhev_datacenter,
+                           'special select'],
+                          ]
+        name = gen_string('alpha')
+        with Session(self.browser) as session:
+            for img_name in valid_data_list():
+                with self.subTest(name):
+                    make_resource(
```
Comment (oshtaier) [inline tests/foreman/ui/test_computeresource_rhev.py]: and here

File: tests/foreman/ui/test_computeresource_rhev.py [hunk 32]
```diff
@@ -176,13 +308,41 @@ def test_positive_add_image_rhev_with_name(self):
...
+                                  session,
+                                  name=name,
+                                  provider_type=FOREMAN_PROVIDERS['rhev'],
+                                  parameter_list=parameter_list
+                                  )
+                parameter_list_img = [
```
Comment (oshtaier) [inline tests/foreman/ui/test_computeresource_rhev.py]: and here

File: tests/foreman/ui/test_computeresource_rhev.py [hunk 33]
```diff
@@ -199,13 +359,42 @@ def test_negative_add_image_rhev_with_invalid_name(self):
         4. Select "New image" , provide it invalid name and valid information.
         5. Select the desired template to create the image from and submit.
 
-        @Assert: The image should not be added to the CR
+        @CaseAutomation: Automated
 
-        @Caseautomation: notautomated
+        @Assert: The image should not be added to the CR
         """
+        parameter_list = [
```
Comment (oshtaier) [inline tests/foreman/ui/test_computeresource_rhev.py]: and here

File: tests/foreman/ui/test_computeresource_rhev.py [hunk 34]
```diff
@@ -199,13 +359,42 @@ def test_negative_add_image_rhev_with_invalid_name(self):
...
+                          ['Username', self.rhev_username, 'field'],
+                          ['Password', self.rhev_password, 'field'],
+                          ['Datacenter', self.rhev_datacenter,
+                           'special select'],
+                          ]
+        name = gen_string('alpha')
+        with Session(self.browser) as session:
+            for img_name in invalid_names_list():
+                with self.subTest(name):
+                    make_resource(
```
Comment (oshtaier) [inline tests/foreman/ui/test_computeresource_rhev.py]: and here

File: tests/foreman/ui/test_computeresource_rhev.py [hunk 35]
```diff
@@ -199,13 +359,42 @@ def test_negative_add_image_rhev_with_invalid_name(self):
...
+                                  session,
+                                  name=name,
+                                  provider_type=FOREMAN_PROVIDERS['rhev'],
+                                  parameter_list=parameter_list
+                                  )
+                parameter_list_img = [
```
Comment (oshtaier) [inline tests/foreman/ui/test_computeresource_rhev.py]: and here

File: tests/foreman/ui/test_computeresource_rhev.py [hunk 36]
```diff
@@ -222,10 +411,27 @@ def test_positive_access_rhev_with_default_profile(self):
         4. Click Compute Profile tab.
         5. Select (3-Large) and submit.
 
-        @Assert: The Compute Resource created and opened successfully
+        @CaseAutomation: Automated
 
-        @Caseautomation: notautomated
+        @Assert: The Compute Resource created and opened successfully
         """
+        parameter_list = [
```
Comment (oshtaier) [inline tests/foreman/ui/test_computeresource_rhev.py]: and here

File: tests/foreman/ui/test_computeresource_rhev.py [hunk 37]
```diff
@@ -222,10 +411,27 @@ def test_positive_access_rhev_with_default_profile(self):
...
+                          ['URL', self.rhev_url, 'field'],
+                          ['Username', self.rhev_username, 'field'],
+                          ['Password', self.rhev_password, 'field'],
+                          ['Datacenter', self.rhev_datacenter,
+                           'special select'],
+                          ]
+        name = gen_string('alpha')
+        with Session(self.browser) as session:
+            make_resource(
```
Comment (oshtaier) [inline tests/foreman/ui/test_computeresource_rhev.py]: and here

File: tests/foreman/ui/test_computeresource_rhev.py [hunk 38]
```diff
@@ -265,10 +470,28 @@ def test_positive_retrieve_rhev_vm_list(self):
         1. Select the created compute resource.
         2. Go to "Virtual Machines" tab.
 
-        @Assert: The Virtual machines should be displayed
+        @CaseAutomation: Automated
 
-        @Caseautomation: notautomated
+        @Assert: The Virtual machines should be displayed
         """
+        parameter_list = [
```
Comment (oshtaier) [inline tests/foreman/ui/test_computeresource_rhev.py]: and here

File: tests/foreman/ui/test_computeresource_rhev.py [hunk 39]
```diff
@@ -265,10 +470,28 @@ def test_positive_retrieve_rhev_vm_list(self):
...
+                          ['URL', self.rhev_url, 'field'],
+                          ['Username', self.rhev_username, 'field'],
+                          ['Password', self.rhev_password, 'field'],
+                          ['Datacenter', self.rhev_datacenter,
+                           'special select'],
+                          ]
+        name = gen_string('alpha')
+        with Session(self.browser) as session:
+            make_resource(
```
Comment (oshtaier) [inline tests/foreman/ui/test_computeresource_rhev.py]: and here

**Comment by sghai** [comment]
@san7ket : hey, Could you please make changes as per review comments and run these tests against nightly, if possible.

**Comment by san7ket** [comment]
All the comments addressed with latest commit. 
Also, not able to test this with nightly as , currently we just support RHEV and the support for oVirt Compute resource provider will be added later.

**Comment by sghai** [comment]
In upstream, for rhev CR provider name is `oVirt`, however in downstream it is 'RHEV`. We may need to come up with some implementation to support both by first fetching available providers from server and accordingly using that or see if we have some other ways to do the same.   Let's tackle this in another PR.

@san7ket: Please open a new issue for this implementation.

**Comment by sghai** [comment]
ACK for this PR.

@oshtaier : PR looks good to me. I would like you to please take another look quickly and see if we are good to merge this. thanks.

File: tests/foreman/ui/test_computeresource_rhev.py [hunk 40]
```diff
@@ -199,13 +352,41 @@ def test_negative_add_image_rhev_with_invalid_name(self):
...
+                    ['Name', img_name],
+                    ['Operatingsystem', self.rhev_img_os],
+                    ['Architecture', self.rhev_img_arch],
+                    ['Username', self.rhev_img_user],
+                    ['Password', self.rhev_img_pass],
+                    ['uuid', self.rhev_img_name],
+                ]
+                self.compute_resource.add_image(name, parameter_list_img)
+                self.assertIsNotNone(
+                        self.compute_resource.wait_until_element(
```
Comment (oshtaier) [inline tests/foreman/ui/test_computeresource_rhev.py]: nitpick: here, it seems, we need to move `wait` block to the left on 4 spaces

**Comment by oshtaier** [comment]
ACK pending test results for any supported build. Sanket addressed a lot of comments, so just to make sure that we not break anything

**Comment by san7ket** [comment]
nosetests tests.foreman.ui.test_computeresource_rhev
..S.....SSSS...
----------------------------------------------------------------------
Ran 15 tests in 692.502s

OK (SKIP=5)

**Comment by oshtaier** [comment]
ACK

**Comment by oshtaier** [comment]
Please rebase

**Comment by sghai** [comment]
ACK..Merging now

**Pull request merged**